masonry/src/text/layout.rs

```diff
@@ -501,6 +516,89 @@ impl<T: AsRef<str> + Eq> TextLayout<T> {
             &self.layout,
         );
     }
+
+    pub fn accessibility(&mut self, ctx: &mut AccessCtx, parent_node: &mut NodeBuilder) {
+        self.assert_rebuilt("accessibility");
+
+        let text = self.text.as_ref();
+        let mut ids = HashSet::<NodeId>::new();
+
+        for (line_index, line) in self.layout.lines().enumerate() {
+            let mut last_node: Option<(NodeId, NodeBuilder)> = None;
+
+            for (run_index, run) in line.runs().enumerate() {
+                let run_path = (line_index, run_index);
+                let id = self
+                    .access_ids_by_run_path
+                    .get(&run_path)
+                    .copied()
+                    .unwrap_or_else(|| {
+                        let id = NodeId::from(WidgetId::next());
+                        self.access_ids_by_run_path.insert(run_path, id);
+                        self.run_paths_by_access_id.insert(id, run_path);
+                        id
+                    });
+                ids.insert(id);
+                let mut node = NodeBuilder::new(Role::InlineTextBox);
+
+                if let Some((last_id, mut last_node)) = last_node.take() {
+                    last_node.set_next_on_line(id);
+                    node.set_previous_on_line(last_id);
+                    ctx.tree_update.nodes.push((last_id, last_node.build()));
+                    parent_node.push_child(last_id);
```

Comment:
This could probably be done in the caller. `TextLayout::Accessibility` could return a `Vec<NodeBuilder>`.

Reply:
I don't like returning a `Vec` because that requires an intermediate allocation. I'll implement Daniel's suggestion of passing in the `TreeUpdate` rather than the `AccessCtx`.